### PR TITLE
Database setup

### PR DESCRIPTION
**Pull request by JordanLindley**
### Implement `deleteNote`, `updateNote`, `readNote` for CRUD operations

Import into router functions in database setup _index.ts_. Need to rework `updateNote` to avoid bug of "body" or "title" reverting to "null" if left undefined by user.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +29,9 @@
     "nodemon": "^2.0.21",
     "ts-node": "^10.9.1",
     "typescript": "^4.9.3",
-    "vite": "^4.1.0"
+    "vite": "^4.1.0",
+    "@types/pg": "^8.6.6",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.9.5"
```
Comment (ScottLindley) [inline package.json]: Let's remove lines 33 and 34. They look like duplicates.

File: src/server/notes/index.ts [hunk 2]
```diff
@@ -1,37 +1,67 @@
...
+  const res = await client.query<Note>(`INSERT INTO notes (owner, title, body) VALUES ($1, $2, $3);`, [owner, title, body]);
+  await client.end();
+
+  return res.rows[0];
+}
+
+export async function deleteNote(id:string): Promise<void> {
+  const client = await getClient();
+
+  const res = await client.query<Note>(`DELETE FROM notes WHERE id = '${id}';`);
```
Comment (ScottLindley) [inline src/server/notes/index.ts]: Prefer using parameterized queries to guard against SQL injection attacks.

```ts
const res = await client.query<Note>(`DELETE FROM notes WHERE id = $1;`, [id]);
```

File: src/server/notes/index.ts [hunk 3]
```diff
@@ -1,37 +1,67 @@
...
+  const res = await client.query<Note>(`UPDATE notes SET title = $1, BODY = $2 WHERE id = $3;`, [title, body, noteID]);
+  await client.end();
+
+  return res.rows[0];
 }
+
+export async function readNote(id:string): Promise<Note> {
+  const client = await getClient();
+
+  const res = await client.query<Note>(`SELECT * FROM notes WHERE id = '${id}';`);
```
Comment (ScottLindley) [inline src/server/notes/index.ts]: Prefer using parameterized queries to guard against SQL injection attacks.

```ts
const res = await client.query<Note>(`SELECT * FROM notes WHERE id = $1;`, [id]);
```

**Review by ScottLindley — APPROVED** [review]
Left some comments. Feel free to address them with another commit, or we can fix them in a follow up PR.

File: src/server/notes/index.ts [hunk 4]
```diff
@@ -1,37 +1,67 @@
...
-  updatedAt: Date;
 };
 
-export type User = {
-  id: string;
-  email: string;
-};
+async function getClient() {
+  const client = new Client({
+    user: 'jordanlindley',
```
Comment (ScottLindley) [inline src/server/notes/index.ts:13]: Note: this will only work on your machine, but I can fix this in a follow up PR.

File: package.json (hunk 1 above)
Comment (JordanLindley) [reply package.json]: Removed

File: src/server/notes/index.ts (hunk 2 above)
Comment (JordanLindley) [reply src/server/notes/index.ts]: Updated!

File: src/server/notes/index.ts (hunk 3 above)
Comment (JordanLindley) [reply src/server/notes/index.ts]: Updated!

**Pull request merged**